net/httpclient.go

```diff
@@ -233,8 +234,12 @@ func NewTransport(options Options) *Transport {
 	if options.ExpectContinueTimeout == 0 {
 		options.ExpectContinueTimeout = options.Timeout
 	}
+	if options.Proxy == nil {
```

Comment:
This would not allow to create transport with `nil` proxy like it was before. 
Should `options.Proxy` be rather passes into `Transport` as is and configured on authClient side?
https://github.com/zalando/skipper/blob/9f8c62c12b9edb6d2f6147221a178383f8f3a4d6/filters/auth/authclient.go#L47-L54

Reply:
I think it‘s great to act similar to the default Go and you can easily pass a noop func with that signature if you want.
As another thought of why it’s fine like this: If someone starts it with the environment variables set she will likely want this as default. We could also set default somewhere else, but I think right now it’s great like this, because everyone will just pass nothing and expect it to work.

Reply:
Ok, not a big deal, the proxy could be still disabled by passing `func(*http.Request) (*url.URL, error) {return nil, nil}`
> // If Proxy is nil or returns a nil *URL, no proxy is used.